crates/cli-support/src/js.rs

```diff
@@ -227,23 +229,59 @@ impl<'a> Context<'a> {
             self.footer.push_str(&format!("wasm = require('./{}_bg');",
                                           module_name));
             format!("var wasm;")
+        } else if self.config.no_modules {
+            format!("
+                return fetch('{module}_bg.wasm')
+                    .then(response => response.arrayBuffer())
+                    .then(buffer => WebAssembly.instantiate(buffer, __js_exports))
+                    .then(({{instance}}) => {{
+                        wasm = instance.exports;
+                        return wasm;
+                    }})
+                    .catch(error => {{
+                        console.log('Error loading wasm module `{module}`:', error);
+                        throw error;
+                    }});
+            ", module = module_name)
         } else {
             format!("import * as wasm from './{}_bg';", module_name)
         };
 
-        let js = format!("
-            /* tslint:disable */
-            {import_wasm}
-            {imports}
-
-            {globals}
-            {footer}
-        ",
-            import_wasm = import_wasm,
-            globals = self.globals,
-            imports = self.imports,
-            footer = self.footer,
-        );
+        let js = if self.config.no_modules {
+            format!("
+                (function (root, factory) {{
+                    if (typeof define === 'function' && define.amd) {{
+                        define([], factory);
+                    }} else {{
+                        root.{module} = factory();
+                    }}
+                }}(typeof self !== 'undefined' ? self : this, function() {{
+                    let wasm;
+                    const __js_exports = {{}};
+                    const __exports = {{}};
+                    {globals}
+                    __js_exports['./{module}'] = __exports;
+                    {import_wasm}
+                }}))
```

Comment:
I think this preamble can be simplified quite a bit, right? If we're only working with one local module I dont think we need the extra function scopes and such?

Reply:
This preamble would be useful if someone else imports it as a library in js but then again they should rather use other builds.